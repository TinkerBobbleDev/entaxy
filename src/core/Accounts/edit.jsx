import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'
import AccountForm from './form'
import { updateAccount, deleteAccount } from '../../store/accounts/actions'
import confirm from '../../util/confirm'

const mapDispatchToProps = {
  handleSave: account => updateAccount(account),
  handleDelete: account => deleteAccount(account)
}

export const EditAccountComponent = ({
  history,
  match,
  handleSave,
  handleDelete
}) => {
  const onSave = (account) => {
    handleSave(account)
    history.push(`/accounts/${account.id}/transactions`)
  }

  const onDelete = (account) => {
    confirm('Delete selected account?', 'Are you sure?').then(() => {
      handleDelete(account)
      history.push('/dashboard')
    })
  }

  const onCancel = () => {
    history.push(`/accounts/${match.params.accountId}/transactions`)
  }

  return (
    <AccountForm
      accountId={match.params.accountId}
      handleSave={onSave}
      handleDelete={onDelete}
      handleCancel={onCancel}
    />
  )
}

EditAccountComponent.propTypes = {
  history: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
  handleSave: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired
}

export default connect(null, mapDispatchToProps)(withRouter(EditAccountComponent))
