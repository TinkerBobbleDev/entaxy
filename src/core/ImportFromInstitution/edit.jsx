import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ImportFromInstitutionForm from './form'
import { updateAccountGroup, deleteAccountGroup } from '../../store/accounts/actions'
import confirm from '../../util/confirm'

const mapDispatchToProps = {
  updateAccountGroup: (institution, accountGroup, accounts) => (
    updateAccountGroup(institution, accountGroup, accounts)
  ),
  deleteAccountGroup: accountGroup => deleteAccountGroup(accountGroup)
}

export class EditImportFromInstitutionComponent extends React.Component {
  handleSave = async (accountGroup, accounts) => {
    const { institution } = this.props.match.params
    this.props.updateAccountGroup(institution, accountGroup, accounts)
    this.props.history.push('/dashboard')
  }

  handleDelete = (accountGroup) => {
    const { accountIds } = accountGroup
    const { institution } = this.props.match.params
    confirm(`Delete all the ${accountIds.length} accounts connected to ${institution}?`, 'Are you sure?')
      .then(async () => {
        await this.props.deleteAccountGroup(accountGroup)
        this.props.history.push('/dashboard')
      })
  }

  handleCancel = () => {
    this.props.history.push('/dashboard')
  }

  render() {
    const { institution, groupId } = this.props.match.params
    return (
      <ImportFromInstitutionForm
        groupId={groupId}
        handleSave={this.handleSave}
        handleCancel={this.handleCancel}
        handleDelete={this.handleDelete}
        institution={institution}
      />
    )
  }
}

EditImportFromInstitutionComponent.propTypes = {
  history: PropTypes.object.isRequired,
  updateAccountGroup: PropTypes.func.isRequired,
  deleteAccountGroup: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired
}

export default connect(null, mapDispatchToProps)(EditImportFromInstitutionComponent)
