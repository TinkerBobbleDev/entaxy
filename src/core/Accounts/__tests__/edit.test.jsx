import React from 'react'
import renderer from 'react-test-renderer'
import { EditAccountComponent } from '../edit'

jest.mock('../form', () => 'AccountForm')

describe('New Account', () => {
  const mochHandleSave = jest.fn()
  const mochHandleCancel = jest.fn()
  const mochHandleDelete = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    const component = renderer.create((
      <EditAccountComponent
        handleSave={mochHandleSave}
        handleCancel={mochHandleCancel}
        handleDelete={mochHandleDelete}
        history={{}}
        match={{ params: { accountId: 1 } }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
