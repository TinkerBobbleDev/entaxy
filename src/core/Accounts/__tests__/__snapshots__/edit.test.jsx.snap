// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Account matches snapshot 1`] = `
<AccountForm
  accountId={1}
  handleCancel={[Function]}
  handleDelete={[Function]}
  handleSave={[Function]}
/>
`;
